### ISSUE TITLE
Remote: SSH Extension doesn't work when config is present

### ISSUE THREAD

**Issue by SleeplessByte** [microsoft/vscode-remote-release#5630]
I don't know if this is a bug in the extension or VS Code itself. Cross linking <https://github.com/microsoft/vscode/issues/133737>.

----

Using the official SSH extensions, since the last update of insiders, anything I try to connect to shows:

> Bad owner or permissions on C:\\Users\\Derk-Jan/.ssh/config
> The process tried to write to a nonexistent pipe.
> 

The configuration file can be opened when the connection fails (Open SSH Configuration File) and there the path shows correctly as `C:\Users\Derk-Jan\.ssh\config`. It is also openable, without permission issues.

VS Code version: Code - Insiders 1.61.0-insider (8d30d4d922dda917d22f05a81f2ab4fad7349779, 2021-09-24T09:00:59.405Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No
Fetching remote diagnostics for 'SSH' failed.
Fetching remote diagnostics for 'SSH' failed.

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6800K CPU @ 3.40GHz (12 x 3398)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.91GB (17.84GB free)|
|Process Argv|--crash-reporter-id b89f360b-5a82-470c-9730-8c9124c84a81|
|Screen Reader|no|
|VM|0%|

Fetching remote diagnostics for 'SSH' failed.

Fetching remote diagnostics for 'SSH' failed.
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30291494
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30291487
pythontb:30258533
pythonvspyt551cf:30291413
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
pythonvsuse255:30319630
vscod805cf:30301675
pythonvspyt200:30323110
vsccppwt:30364496
pythonvssor306:30340298
bridge0708:30335490
pygetstartedt2:30353727
dockerwalkthru:30364418
bridge0723:30353136
pythonrunftest32:30365365
pythonf5test824:30361779
javagetstartedt:30350119
pythonvspyt187:30365360
pydsgst2:30361790
vssur157:30367808

```

</details>

-----

Reproducible with both VS Code and VS Code - Insiders. Worked before I updated today.

As a default, when this happens, I try to reproduce in non-insiders. I was able to reproduce the _exact_ same issue in non-insiders.

- I've reinstalled VSCode (non-insiders)
- I've reinstalled the Remote SSH extension
- I've reinstalled the Windows OpenSSH Client

I was able to reproduce the issue (connect to a host you've connected to before) and get the same issue.

I was also able to confirm that `ssh-add -l` shows it's running the agent with the correct identity.
I was also able to confirm that a blank call to `ssh -T ... -D ...` does actually work.

## Workaround

Go into `C:\Users\Derk-Jan\.ssh\` and rename `config` to `config.bak`. Now everything works as expected. As soon as I create a config again (through VS Code), everything breaks, even connecting to a host _NOT_ in the config.

Let me know if there is _anything_ I can do to provide more context.

**Comment by roblourens** [comment]
Can you share the log from the Remote-SSH output channel?

**Comment by SleeplessByte** [comment]
```
[19:29:26.059] Log Level: 2
[19:29:26.062] remote-ssh@0.65.8
[19:29:26.062] win32 x64
[19:29:26.063] SSH Resolver called for "ssh-remote+7b22686<snip>6f74227d", attempt 1
[19:29:26.063] "remote.SSH.useLocalServer": false
[19:29:26.064] "remote.SSH.showLoginTerminal": false
[19:29:26.064] "remote.SSH.remotePlatform": {"something.something":"linux","thing.i.connect.to":"linux"}
[19:29:26.065] "remote.SSH.path": undefined
[19:29:26.065] "remote.SSH.configFile": undefined
[19:29:26.065] "remote.SSH.useFlock": true
[19:29:26.065] "remote.SSH.lockfilesInTmp": false
[19:29:26.066] "remote.SSH.localServerDownload": auto
[19:29:26.066] "remote.SSH.remoteServerListenOnSocket": false
[19:29:26.066] "remote.SSH.showLoginTerminal": false
[19:29:26.066] "remote.SSH.defaultExtensions": []
[19:29:26.067] "remote.SSH.loglevel": 2
[19:29:26.067] SSH Resolver called for host: username@thing.i.connect.to
[19:29:26.069] Setting up SSH remote "thing.i.connect.to"
[19:29:26.092] Using commit id "7f6ab5485bbc008386c4386d08766667e155244e" and quality "stable" for server
[19:29:26.095] Install and start server if needed
[19:29:26.101] Checking ssh with "ssh -V"
[19:29:26.149] > OpenSSH_for_Windows_8.1p1, LibreSSL 
[19:29:26.149] > 3.0.2

[19:29:26.154] Running script with connection command: ssh -T -D 52938 "username@thing.i.connect.to" bash
[19:29:26.156] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[19:29:26.383] > ]0;C:\WINDOWS\System32\cmd.exe
[19:29:26.383] Got some output, clearing connection timeout
[19:29:26.406] > Bad owner or permissions on C:\\Users\\Derk-Jan/.ssh/config
> The process tried to write to a nonexistent pipe.
>         
[19:29:27.680] "install" terminal command done
[19:29:27.680] Install terminal quit with output:         
[19:29:27.680] Received install output:         
[19:29:27.681] Resolver error: Error: 
	at Function.Create (c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:397041)
	at c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:395057
	at Object.t.handleInstallOutput (c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:395620)
	at Object.t.tryInstall (c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:489314)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:453110
	at async Object.t.withShowDetailsEvent (c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:456460)
	at async Object.t.resolve (c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:454189)
	at async c:\Users\Derk-Jan\.vscode\extensions\ms-vscode-remote.remote-ssh-0.65.8\out\extension.js:1:526942
[19:29:27.685] ------
```

The following command works as expected:

```shell
ssh -T -D 52938 "username@thing.i.connect.to"
```

**Comment by SleeplessByte** [comment]
Is there anything else I can do to help?

**Comment by roblourens** [comment]
There is also a utility to help fix permissions here https://github.com/PowerShell/Win32-OpenSSH/wiki/OpenSSH-utility-scripts-to-fix-file-permissions

**Comment by SleeplessByte** [comment]
I managed to run the fix file permissions script by extracting the scripts to my install location and unblocking them (as they're unsigned "remote" scripts).

The first time I ran it, it told me this:

```text
Need to remove the inheritance before repair the rules.
Shall I remove the inheritace?

Y

Inheritance is removed from '~\.ssh\config'.
```

The second time I ran it, it showed me this:

![image](https://user-images.githubusercontent.com/1964376/137610593-df525678-7ac6-4c96-a545-0f7e6915ac4e.png)

My VS Code started working again (despite that PowerShell error). 

Whilst _my_ problem is solved, I want to re-iterate that I have not touched the permissions of these files, moved them around, changed them in any way manually, so I don't know if this is safe to close, but I guess if someone else runs into this, it's worth a try to run those scripts first.

Thanks Rob.
